src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeGenerator.java

```diff
@@ -955,16 +984,16 @@ private void generate() {
     }
 
     // Checks if the Op.Result is used more than once in operands and block arguments
-    private static boolean moreThanOneUse(Op.Result res) {
-        return res.uses().stream().flatMap(u ->
+    private static boolean moreThanOneUse(Value val) {
```

Comment:
Memo to self - we should have something in the API. More generally, liveness analysis of values might help formalize when a value is no longer used and its slot can be reused with another value. I suspect generating bytecode in from models in pure SSA might result use of more locals than actually needed.